forge/forge/agent/components.py

```diff
@@ -39,18 +39,24 @@ def run_after(self: T, *components: type[AgentComponent] | AgentComponent) -> T:
 class ConfigurableComponent(ABC, Generic[C]):
     """A component that can be configured with a Pydantic model."""
 
-    def __init__(self, config: Optional[C] = None):
+    config_class: ClassVar
```

Comment:
```suggestion
    config_class: ClassVar[type[BaseModel]]
```

Reply:
Thanks for explanation! I used `config_class: ClassVar[type[C]]  # type: ignore`
Also renamed
`T` -> `AC` (`AgentComponent`)
`C` -> `BM` (`BaseModel`)